Doc/c-api/init.rst

```diff
@@ -7,6 +7,126 @@
 Initialization, Finalization, and Threads
 *****************************************
 
+.. _pre-init-safe:
+
+Before Python Initialization
+============================
+
+In an application embedding  Python, the :c:func:`Py_Initialize` function must
+be called before using any other Python/C API functions; with the exception of
+a few functions and flags listed below.
+
+The following functions can be safetely called before Python is initialized:
+
+* Functions:
+
+  * :c:func:`PyMem_GetAllocator`
+  * :c:func:`PyMem_SetAllocator`
+  * :c:func:`PyMem_SetupDebugHooks`
+  * :c:func:`PyObject_GetArenaAllocator`
+  * :c:func:`PyObject_SetArenaAllocator`
+  * :c:func:`Py_DecodeLocale`
+  * :c:func:`Py_GetProgramName`
+  * :c:func:`Py_SetPath`
+  * :c:func:`Py_SetProgramName`
+  * :c:func:`Py_SetPythonHome`
+  * :c:func:`Py_SetStandardStreamEncoding`
+
+* Memory allocators:
+
+  * :c:func:`PyMem_RawMalloc`
+  * :c:func:`PyMem_RawRealloc`
+    :c:func:`PyMem_RawCalloc`
+  * :c:func:`PyMem_RawFree`
+  * :c:func:`PyMem_Malloc`
+  * :c:func:`PyMem_Realloc`
+  * :c:func:`PyMem_Calloc`
+  * :c:func:`PyMem_Free`
+  * :c:func:`PyObject_Malloc`
+  * :c:func:`PyObject_Realloc`
+  * :c:func:`PyObject_Calloc`
+  * :c:func:`PyObject_Free`
+
+The following flags can also be set:
+
+* :c:data:`Py_BytesWarningFlag`
+* :c:data:`Py_DebugFlag`
+* :c:data:`Py_InspectFlag`
+* :c:data:`Py_InteractiveFlag`
+* :c:data:`Py_IsolatedFlag`
+* :c:data:`Py_OptimizeFlag`
+* :c:data:`Py_DontWriteBytecodeFlag`
+* :c:data:`Py_NoUserSiteDirectory`
+* :c:data:`Py_NoSiteFlag`
+* :c:data:`Py_UnbufferedStdioFlag`
+* :c:data:`Py_VerboseFlag`
+* :c:data:`Py_QuietFlag`
+* :c:data:`Py_IgnoreEnvironmentFlag`
+
+
+Global configuration variables
+==============================
+
+Python has variables for the global configuration to control different features
+and options. By default, these flags are controlled by :ref:`command line
+options <using-on-interface-options>`.
+
+.. :c:data: Py_BytesWarningFlag
+
+  The :option:`-b` option.
```

Comment:
Wrong indentation (should be 3 spaces).

Document what is the value if use `-bb`. And the same for `-OO` below.

Reply:
Done